design/tls-client-verification.md

```diff
@@ -0,0 +1,125 @@
+# External client certificate validation (mutual TLS authentication)
+
+Status: Draft
+
+This document outlines how to add support for authentication of external
+clients by validating their client certificates.
+
+## Goals
+
+- Allow Contour to be used to protect the backend services from access of
+  unauthorized external clients.
+- Allow authentication of external clients by having Envoy validate that
+  client certificates are signed by trusted CA (Certificate Authority)
+- Allow configuration of trusted CA certificate(s) for validating the client
+  certificates.
+
+## Non Goals
+
+- Fine grained authorization on level of individual request
+  (example of non-goal: only client X can access resource Y on backend
+  service Z)
+
+## Background
+
+In TLS (and HTTPS) often only the client authenticates the server.  TLS
```

Comment:
Please remove all the words about mutual TLS authentication, as I understand it this proposal is only about client side certificate authentication. If you want to add mutual TLS as a non goal, please do so in the previous section.

Reply:
I have now removed references to mutual TLS authentication and mTLS.